src/main/java/by/com/epam/repository/RepositoryUser.java

```
@@ -0,0 +1,47 @@
+package by.com.epam.repository;
+
+
+import by.com.epam.comporator.ComparatorFactory;
+import by.com.epam.model.BookModel;
+import by.com.epam.model.BookTagType;
+import by.com.epam.specification.Specification;
+import by.com.epam.specification.SpecificationFactory;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class RepositoryUser implements BookRepository {
+    private final List<BookModel> bookModels;
+
+    public RepositoryUser() {
+        bookModels = new ArrayList<BookModel>();
+    }
+
+    @Override
+    public void addBook(BookModel bookModel) {
```

according to the task you need to throw an exception if the object exists